packages/functions/src/dataTriggers.ts

```diff
@@ -338,3 +338,100 @@ export const createPublicOrgInfo = functions
     );
     await publicOrgInfoDocRef.set(updates, { merge: true });
   });
+
+export const createAttendedSlotOnAttendance = functions
+  .region(__functionsZone__)
+  .firestore.document(
+    `${Collection.Organizations}/{organization}/${OrgSubCollection.Attendance}/{slotId}`
+  )
+  .onWrite(async (change, context) => {
+    const { organization, slotId } = context.params as Record<string, string>;
+
+    const db = admin.firestore();
+
+    const currentAttendanceData = change.after.data() as SlotAttendnace;
+    const previousAttendanceData = change.before.data() as SlotAttendnace;
+
+    const hasCurrentAttendances =
+      change.after.exists &&
+      Object.keys(currentAttendanceData.attendances).length;
+    const hasPreviousAttendances =
+      change.before.exists &&
+      Object.keys(previousAttendanceData.attendances).length;
+
+    const currentAttendances = hasCurrentAttendances
+      ? currentAttendanceData.attendances
+      : {};
+    const previousAttendances = hasPreviousAttendances
+      ? previousAttendanceData.attendances
+      : {};
+
+    const ids = [
+      ...new Set([
+        ...Object.keys(currentAttendances),
+        ...Object.keys(previousAttendances),
+      ]),
+    ];
+
+    const idsMap = ids.map((id) => {
+      /** @TODO remove this if firestore rule works */
+      const hasPreviousBooking =
+        previousAttendances[id] && previousAttendances[id].bookedInterval;
+      const isBooking =
+        currentAttendances[id] && currentAttendances[id].bookedInterval;
+
+      if (hasPreviousBooking || isBooking) return { [id]: "noUpdate" };
+
+      return currentAttendanceData.attendances[id]
+        ? previousAttendanceData.attendances[id]
+          ? currentAttendances[id].attendedInterval ===
+            previousAttendances[id].attendedInterval
+            ? { [id]: "noUpdate" }
+            : { [id]: "update" }
+          : { [id]: "update" }
+        : { [id]: "delete" };
```

Comment:
I find nested ternary's like this very difficult to decipher: I've been reading this for a few minutes and it still isn't very clear to me what going in and causes what to come out. I would suggest that if it goes deeper than 2 "?"'s, try to compose the end result from individual conditional statements.